python-shell/src/gafferpy/gaffer_operations.py

```diff
@@ -2313,6 +2316,16 @@ def __init__(self,
                     op = JsonConverter.from_json(op)
                 self.operations.append(op)
 
+        if conditional is not None:
```

Comment:
(Low Pri)  I'm kinda conflicted on whether to suggest you just replace all this with:

```
try:
  self.conditional = JsonConverter.from_json(conditional, Conditional)
except Exception:
  pass
```

... which I think is supposedly more "pythonic" and avoids re-using the 'conditional' variable.  Although being a Java developer by heart many of my instincts want to write it your way.

Happy to approve as-is, just some food for thought

Reply:
I completely agree, this has bugged me for a while but I just went with the precedent for this fix. I have opened a new ticket to fix this in the whole gafferpy code base (#896), but will merge this one as is. Cheers for the review!